src/coreclr/System.Private.CoreLib/src/System/Threading/Interlocked.CoreCLR.cs

```diff
@@ -127,6 +127,27 @@ public static int CompareExchange(ref int location1, int value, int comparand)
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern int CompareExchange32(ref int location1, int value, int comparand);
 
+        // This is used internally in cases where having a managed
+        // ref to the location is unsafe (Ex: it is the syncblock of a pinned object).
+        // The intrinsic expansion for this overload is exactly the same as for the `ref int`
+        // variant and will go on the same path since expansion is triggered by the name and
+        // return type of the method.
+        // The important part is avoiding `ref *location` in the unexpanded scenario, like
+        // in a case when compiling the Debug flavor of the app.
```

Comment:
```suggestion
        // The important part is avoiding `ref *location` that is reported as byref to the GC.
```
I do not think the expanded vs. non-expanded matters. If the expanded version reports the byref to the GC, it would be a problem as well.

It would be nice to fix NAOT comment to match.